legend-engine-xt-sql-postgres-server/src/main/java/org/finos/legend/engine/postgres/Session.java

```diff
@@ -103,16 +105,31 @@ public void parse(String statementName, String query, List<Integer> paramTypes)
     private SessionHandler getSessionHandler(String query)
     {
         SqlBaseParser parser = SQLGrammarParser.getSqlBaseParser(query, "query");
-        List<QualifiedName> qualifiedNames = EXTRACTOR.visitSingleStatement(parser.singleStatement());
-
-        boolean isMetadataQuery = qualifiedNames.stream().flatMap(i -> i.parts.stream()).anyMatch(SystemSchemas::contains);
-        if (isMetadataQuery)
+        try
         {
-            return metaDataSessionHandler;
+            List<QualifiedName> qualifiedNames = EXTRACTOR.visitSingleStatement(parser.singleStatement());
```

Comment:
Is there a way to differentiating at the single statement type level if it's a select statement of a different type of statement (before performing the visit), e.g. instance of Select vs. FunctionCall